File: packages/csharp/src/Catalog/Carriers.cs
// CATALOG-GEN: do not hand-edit; rerun packages/csharp/scripts/gen-catalog.mjs.
//
// Source data:
//   - insurance/v2_products.json

using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;

namespace Isa.Sdk.Catalog;

/// <summary>Public metadata for a single carrier. Today's catalog does not
/// expose per-carrier licensure data; <c>States</c> is intentionally omitted.</summary>
public sealed record ProductCarrierMetadata(
    string DisplayName,
    IReadOnlyList<Product> Products);

/// <summary>Catalog API for carriers.</summary>
public static class ProductCarriers
{
    private static readonly IReadOnlyDictionary<string, ProductCarrierMetadata> CARRIERS = new ReadOnlyDictionary<string, ProductCarrierMetadata>(new Dictionary<string, ProductCarrierMetadata>
    {
        ["aetna"] = new ProductCarrierMetadata("Aetna", new[] { Products.Fex.AetnaAccendo, Products.Fex.AetnaProtectionSeries, Products.Medsup.AetnaMedsup }),
        ["aetna-accendo"] = new ProductCarrierMetadata("Aetna Accendo", new[] { Products.Medsup.AetnaAccendoMedsup }),
        ["aflac"] = new ProductCarrierMetadata("Aflac", new[] { Products.Fex.AflacFinalExpense }),
        ["american-amicable"] = new ProductCarrierMetadata("American Amicable", new[] { Products.Fex.AmericanAmicableClearChoice, Products.Fex.AmericanAmicableDignitySolutions, Products.Fex.AmericanAmicableGoldenSolution, Products.Fex.AmericanAmicableInnovativeSolutions, Products.Fex.AmericanAmicablePlatinumSolutionLegacyPlan, Products.Fex.AmericanAmicableSeniorChoice, Products.Fex.AmericanAmicableTribute, Products.Term.AmericanAmicableEasyTerm, Products.Term.AmericanAmicableHomeProtector, Products.Term.AmericanAmicableTermMadeSimple }),
        ["american-home-life"] = new ProductCarrierMetadata("American Home Life", new[] { Products.Fex.AmericanHomeLifeGuidestar, Products.Fex.AmericanHomeLifePatriotSeries }),
        ["americo"] = new ProductCarrierMetadata("Americo", new[] { Products.Fex.AmericoEaglePremier, Products.Term.AmericoHmsPlus }),
        ["ameritas"] = new ProductCarrierMetadata("Ameritas", new[] { Products.Term.AmeritasFlxLivingBenefitsTerm, Products.Term.AmeritasValuePlusTerm }),
        ["baltimore-life"] = new ProductCarrierMetadata("Baltimore Life", new[] { Products.Fex.BaltimoreLifeIprovide, Products.Fex.BaltimoreLifeSilverGuard }),
        ["banner"] = new ProductCarrierMetadata("Banner", new[] { Products.Term.BannerOpterm }),
        ["betterlife"] = new ProductCarrierMetadata("BetterLife", new[] { Products.Fex.BetterlifeFinalExpense, Products.Preneed.BetterlifeSinglePremium }),
        ["centrian"] = new ProductCarrierMetadata("Centrian", new[] { Products.Fex.CentrianLivingLegacy }),
        ["cica-life"] = new ProductCarrierMetadata("CICA Life", new[] { Products.Fex.CicaLifeSuperiorChoice }),
        ["cigna"] = new ProductCarrierMetadata("Cigna", new[] { Products.Fex.CignaIndividualWholeLife }),
        ["combined"] = new ProductCarrierMetadata("Combined", new[] { Products.Fex.CombinedGenerationalLife }),
        ["corebridge"] = new ProductCarrierMetadata("Corebridge", new[] { Products.Fex.CorebridgeGiwl, Products.Fex.CorebridgeSimplinowLegacy, Products.Term.CorebridgeSelectATerm }),
        ["emc"] = new ProductCarrierMetadata("EMC", new[] { Products.Fex.EmcEasylife }),
        ["everest-ia-american"] = new ProductCarrierMetadata("Everest IA American", new[] { Products.Fex.EverestIaAmericanAdvantage50Plus }),
        ["family-benefit-life"] = new ProductCarrierMetadata("Family Benefit Life", new[] { Products.Fex.FamilyBenefitLifeGoldenEagle }),
        ["fidelity-life"] = new ProductCarrierMetadata("Fidelity Life", new[] { Products.Fex.FidelityLifeRapidecision, Products.Fex.FidelityLifeRapidecisionSeniorLife, Products.Term.FidelityLifeInstabrainPureTerm, Products.Term.FidelityLifeInstabrainTerm, Products.Term.FidelityLifeInstaterm }),
        ["first-guaranty-insurance"] = new ProductCarrierMetadata("First Guaranty Insurance", new[] { Products.Fex.FirstGuarantyInsuranceSecurityCare }),
        ["foresters"] = new ProductCarrierMetadata("Foresters", new[] { Products.Fex.ForestersPlanRight, Products.Term.ForestersStrongFoundation, Products.Term.ForestersYourTerm, Products.Term.ForestersYourTermNonMedical }),
        ["gerber"] = new ProductCarrierMetadata("Gerber", new[] { Products.Fex.GerberLife }),
        ["global-atlantic"] = new ProductCarrierMetadata("Global Atlantic", new[] { Products.Preneed.GlobalAtlanticSimpleProtectionPlan }),
        ["gpm"] = new ProductCarrierMetadata("GPM", new[] { Products.Term.GpmQMark }),
        ["gpm-life"] = new ProductCarrierMetadata("GPM Life", new[] { Products.Fex.GpmLifeSecureMark }),
        ["gtl"] = new ProductCarrierMetadata("GTL", new[] { Products.Fex.GtlHeritagePlan, Products.Term.GtlTurboTerm }),
        ["hero-life"] = new ProductCarrierMetadata("Hero Life", new[] { Products.Term.HeroLifeTerm }),
        ["illinois-mutual"] = new ProductCarrierMetadata("Illinois Mutual", new[] { Products.Fex.IllinoisMutualPathProtectorPlus }),
        ["john-hancock"] = new ProductCarrierMetadata("John Hancock", new[] { Products.Term.JohnHancockSimpleTermWithVitality }),
        ["kansas-city-life"] = new ProductCarrierMetadata("Kansas City Life", new[] { Products.Term.KansasCityLifeSignatureTermExpress }),
        ["kskj"] = new ProductCarrierMetadata("KSKJ", new[] { Products.Fex.KskjFinalExpense }),
        ["liberty-bankers"] = new ProductCarrierMetadata("Liberty Bankers", new[] { Products.Fex.LibertyBankersSimpl }),
        ["life-shield"] = new ProductCarrierMetadata("Life Shield", new[] { Products.Fex.LifeShieldSurvivor }),
        ["lincoln"] = new ProductCarrierMetadata("Lincoln", new[] { Products.Term.LincolnLifeelements, Products.Term.LincolnTermaccel }),
        ["manhattan-life"] = new ProductCarrierMetadata("Manhattan Life", new[] { Products.Fex.ManhattanLifeSecureAdvantage, Products.Medsup.ManhattanLifeMedsup }),
        ["mutual-of-omaha"] = new ProductCarrierMetadata("Mutual of Omaha", new[] { Products.Fex.MutualOfOmahaLivingPromise, Products.Medsup.MutualOfOmahaMedsup, Products.Term.MutualOfOmahaTermLifeAnswers, Products.Term.MutualOfOmahaTermLifeExpress }),
        ["nationwide"] = new ProductCarrierMetadata("Nationwide", new[] { Products.Term.NationwideYourlife }),
        ["newbridge"] = new ProductCarrierMetadata("Newbridge", new[] { Products.Fex.NewbridgeFinalExpense }),
        ["north-american"] = new ProductCarrierMetadata("North American", new[] { Products.Term.NorthAmericanAddvantage }),
        ["occidental-life"] = new ProductCarrierMetadata("Occidental Life", new[] { Products.Fex.OccidentalLifeClearChoice, Products.Fex.OccidentalLifeDignitySolutions, Products.Fex.OccidentalLifeGoldenSolution, Products.Fex.OccidentalLifeInnovativeSolutions, Products.Fex.OccidentalLifePlatinumSolutionLegacyPlan, Products.Fex.OccidentalLifeSeniorChoice, Products.Fex.OccidentalLifeTribute }),
        ["oxford-life"] = new ProductCarrierMetadata("Oxford Life", new[] { Products.Fex.OxfordLifeSimplifiedIssue }),
        ["pekin"] = new ProductCarrierMetadata("Pekin", new[] { Products.Fex.PekinWholeLife }),
        ["pioneer-american"] = new ProductCarrierMetadata("Pioneer American", new[] { Products.Fex.PioneerAmericanIndependentAmerican, Products.Fex.PioneerAmericanNorthstarLegacy }),
        ["prosperity"] = new ProductCarrierMetadata("Prosperity", new[] { Products.Term.ProsperityFamilyFreedomTerm }),
        ["protective-life"] = new ProductCarrierMetadata("Protective Life", new[] { Products.Term.ProtectiveLifeClassicChoiceTerm, Products.Term.ProtectiveLifeCustomChoiceTerm }),
        ["prudential"] = new ProductCarrierMetadata("Prudential", new[] { Products.Term.PrudentialEssentialTermPlus, Products.Term.PrudentialEssentialTermValue }),
        ["royal-arcanum"] = new ProductCarrierMetadata("Royal Arcanum", new[] { Products.Fex.RoyalArcanumGraded, Products.Fex.RoyalArcanumSimplifiedIssue }),
        ["royal-neighbors"] = new ProductCarrierMetadata("Royal Neighbors", new[] { Products.Fex.RoyalNeighborsEnsuredLegacy }),
        ["s-usa"] = new ProductCarrierMetadata("S.USA", new[] { Products.Fex.SUsaGoldenPromise }),
        ["sagicor"] = new ProductCarrierMetadata("Sagicor", new[] { Products.Term.SagicorSageTerm }),
        ["sbli"] = new ProductCarrierMetadata("SBLI", new[] { Products.Fex.SbliLivingLegacy, Products.Term.SbliTTerm }),
        ["securico-life"] = new ProductCarrierMetadata("Securico Life", new[] { Products.Fex.SecuricoLifeFinalExpense }),
        ["security-national"] = new ProductCarrierMetadata("Security National", new[] { Products.Fex.SecurityNationalSimpleSecurity }),
        ["senior-life"] = new ProductCarrierMetadata("Senior Life", new[] { Products.Fex.SeniorLifeWholeLife, Products.Term.SeniorLifeTermLife }),
        ["sentinel-security"] = new ProductCarrierMetadata("Sentinel Security", new[] { Products.Fex.SentinelSecurityNewVantage }),
        ["sons-of-norway"] = new ProductCarrierMetadata("Sons of Norway", new[] { Products.Fex.SonsOfNorwayLegacySure, Products.Fex.SonsOfNorwayWholeLife }),
        ["transamerica"] = new ProductCarrierMetadata("TransAmerica", new[] { Products.Fex.TransamericaFeExpressSolution, Products.Fex.TransamericaSolution, Products.Term.TransamericaTrendsetterLb, Products.Term.TransamericaTrendsetterSuper }),
        ["trinity"] = new ProductCarrierMetadata("Trinity", new[] { Products.Fex.TrinityGoldenEagle }),
        ["united-farm-and-family"] = new ProductCarrierMetadata("United Farm And Family", new[] { Products.Fex.UnitedFarmAndFamilyWholeLife }),
        ["united-home-life"] = new ProductCarrierMetadata("United Home Life", new[] { Products.Fex.UnitedHomeLifeWholeLife }),
        ["william-penn"] = new ProductCarrierMetadata("William Penn", new[] { Products.Term.WilliamPennOpterm }),
    });

    /// <summary>Every carrier slug.</summary>
    public static IReadOnlyCollection<string> Values() => new List<string>(CARRIERS.Keys).AsReadOnly();

    /// <summary>Metadata lookup for a carrier slug. Case-insensitive.</summary>
    public static ProductCarrierMetadata Metadata(string carrier)
    {
        if (carrier is null) throw new ArgumentNullException(nameof(carrier));
        var key = carrier.ToLowerInvariant();
        if (!CARRIERS.TryGetValue(key, out var m))
            throw new ArgumentException($"ProductCarriers.Metadata: unknown carrier '{carrier}'", nameof(carrier));
        return m;
    }
}

File: packages/python/src/sah_sdk/catalog/carriers.py
"""Generated catalog module — do not hand-edit; rerun the generator.

Produced by ``packages/python/scripts/gen_catalog.py``.
Regenerate with ``python packages/python/scripts/gen_catalog.py``.
"""
# Source data:
#   - insurance/v2_products.json
from __future__ import annotations

from dataclasses import dataclass

from .products import Product, _FexProducts, _MedsupProducts, _PreneedProducts, _TermProducts


@dataclass(frozen=True, slots=True)
class ProductCarrierMetadata:
    """Public metadata for a single carrier."""

    display_name: str
    products: tuple[Product, ...]
    #: ISO 2-letter state codes the carrier is licensed in. Empty today.
    states: tuple[str, ...]


_CARRIERS: dict[str, ProductCarrierMetadata] = {
    'aetna': ProductCarrierMetadata(display_name='Aetna', products=(_FexProducts.AetnaProtectionSeries, _MedsupProducts.AetnaMedsup, _FexProducts.AetnaAccendo,), states=()),
    'aetna-accendo': ProductCarrierMetadata(display_name='Aetna Accendo', products=(_MedsupProducts.AetnaAccendoMedsup,), states=()),
    'aflac': ProductCarrierMetadata(display_name='Aflac', products=(_FexProducts.AflacFinalExpense,), states=()),
    'american-amicable': ProductCarrierMetadata(display_name='American Amicable', products=(_FexProducts.AmericanAmicableInnovativeSolutions, _FexProducts.AmericanAmicableDignitySolutions, _FexProducts.AmericanAmicableSeniorChoice, _FexProducts.AmericanAmicableClearChoice, _TermProducts.AmericanAmicableHomeProtector, _TermProducts.AmericanAmicableEasyTerm, _FexProducts.AmericanAmicableTribute, _FexProducts.AmericanAmicableGoldenSolution, _TermProducts.AmericanAmicableTermMadeSimple, _FexProducts.AmericanAmicablePlatinumSolutionLegacyPlan,), states=()),
    'american-home-life': ProductCarrierMetadata(display_name='American Home Life', products=(_FexProducts.AmericanHomeLifePatriotSeries, _FexProducts.AmericanHomeLifeGuidestar,), states=()),
    'americo': ProductCarrierMetadata(display_name='Americo', products=(_FexProducts.AmericoEaglePremier, _TermProducts.AmericoHmsPlus,), states=()),
    'ameritas': ProductCarrierMetadata(display_name='Ameritas', products=(_TermProducts.AmeritasValuePlusTerm, _TermProducts.AmeritasFlxLivingBenefitsTerm,), states=()),
    'baltimore-life': ProductCarrierMetadata(display_name='Baltimore Life', products=(_FexProducts.BaltimoreLifeIprovide, _FexProducts.BaltimoreLifeSilverGuard,), states=()),
    'banner': ProductCarrierMetadata(display_name='Banner', products=(_TermProducts.BannerOpterm,), states=()),
    'betterlife': ProductCarrierMetadata(display_name='BetterLife', products=(_PreneedProducts.BetterlifeSinglePremium, _FexProducts.BetterlifeFinalExpense,), states=()),
    'centrian': ProductCarrierMetadata(display_name='Centrian', products=(_FexProducts.CentrianLivingLegacy,), states=()),
    'cica-life': ProductCarrierMetadata(display_name='CICA Life', products=(_FexProducts.CicaLifeSuperiorChoice,), states=()),
    'cigna': ProductCarrierMetadata(display_name='Cigna', products=(_FexProducts.CignaIndividualWholeLife,), states=()),
    'combined': ProductCarrierMetadata(display_name='Combined', products=(_FexProducts.CombinedGenerationalLife,), states=()),
    'corebridge': ProductCarrierMetadata(display_name='Corebridge', products=(_TermProducts.CorebridgeSelectATerm, _FexProducts.CorebridgeSimplinowLegacy, _FexProducts.CorebridgeGiwl,), states=()),
    'emc': ProductCarrierMetadata(display_name='EMC', products=(_FexProducts.EmcEasylife,), states=()),
    'everest-ia-american': ProductCarrierMetadata(display_name='Everest IA American', products=(_FexProducts.EverestIaAmericanAdvantage50Plus,), states=()),
    'family-benefit-life': ProductCarrierMetadata(display_name='Family Benefit Life', products=(_FexProducts.FamilyBenefitLifeGoldenEagle,), states=()),
    'fidelity-life': ProductCarrierMetadata(display_name='Fidelity Life', products=(_TermProducts.FidelityLifeInstabrainTerm, _FexProducts.FidelityLifeRapidecisionSeniorLife, _FexProducts.FidelityLifeRapidecision, _TermProducts.FidelityLifeInstabrainPureTerm, _TermProducts.FidelityLifeInstaterm,), states=()),
    'first-guaranty-insurance': ProductCarrierMetadata(display_name='First Guaranty Insurance', products=(_FexProducts.FirstGuarantyInsuranceSecurityCare,), states=()),
    'foresters': ProductCarrierMetadata(display_name='Foresters', products=(_TermProducts.ForestersStrongFoundation, _TermProducts.ForestersYourTerm, _FexProducts.ForestersPlanRight, _TermProducts.ForestersYourTermNonMedical,), states=()),
    'gerber': ProductCarrierMetadata(display_name='Gerber', products=(_FexProducts.GerberLife,), states=()),
    'global-atlantic': ProductCarrierMetadata(display_name='Global Atlantic', products=(_PreneedProducts.GlobalAtlanticSimpleProtectionPlan,), states=()),
    'gpm': ProductCarrierMetadata(display_name='GPM', products=(_TermProducts.GpmQMark,), states=()),
    'gpm-life': ProductCarrierMetadata(display_name='GPM Life', products=(_FexProducts.GpmLifeSecureMark,), states=()),
    'gtl': ProductCarrierMetadata(display_name='GTL', products=(_FexProducts.GtlHeritagePlan, _TermProducts.GtlTurboTerm,), states=()),
    'hero-life': ProductCarrierMetadata(display_name='Hero Life', products=(_TermProducts.HeroLifeTerm,), states=()),
    'illinois-mutual': ProductCarrierMetadata(display_name='Illinois Mutual', products=(_FexProducts.IllinoisMutualPathProtectorPlus,), states=()),
    'john-hancock': ProductCarrierMetadata(display_name='John Hancock', products=(_TermProducts.JohnHancockSimpleTermWithVitality,), states=()),
    'kansas-city-life': ProductCarrierMetadata(display_name='Kansas City Life', products=(_TermProducts.KansasCityLifeSignatureTermExpress,), states=()),
    'kskj': ProductCarrierMetadata(display_name='KSKJ', products=(_FexProducts.KskjFinalExpense,), states=()),
    'liberty-bankers': ProductCarrierMetadata(display_name='Liberty Bankers', products=(_FexProducts.LibertyBankersSimpl,), states=()),
    'life-shield': ProductCarrierMetadata(display_name='Life Shield', products=(_FexProducts.LifeShieldSurvivor,), states=()),
    'lincoln': ProductCarrierMetadata(display_name='Lincoln', products=(_TermProducts.LincolnTermaccel, _TermProducts.LincolnLifeelements,), states=()),
    'manhattan-life': ProductCarrierMetadata(display_name='Manhattan Life', products=(_MedsupProducts.ManhattanLifeMedsup, _FexProducts.ManhattanLifeSecureAdvantage,), states=()),
    'mutual-of-omaha': ProductCarrierMetadata(display_name='Mutual of Omaha', products=(_TermProducts.MutualOfOmahaTermLifeExpress, _MedsupProducts.MutualOfOmahaMedsup, _TermProducts.MutualOfOmahaTermLifeAnswers, _FexProducts.MutualOfOmahaLivingPromise,), states=()),
    'nationwide': ProductCarrierMetadata(display_name='Nationwide', products=(_TermProducts.NationwideYourlife,), states=()),
    'newbridge': ProductCarrierMetadata(display_name='Newbridge', products=(_FexProducts.NewbridgeFinalExpense,), states=()),
    'north-american': ProductCarrierMetadata(display_name='North American', products=(_TermProducts.NorthAmericanAddvantage,), states=()),
    'occidental-life': ProductCarrierMetadata(display_name='Occidental Life', products=(_FexProducts.OccidentalLifeDignitySolutions, _FexProducts.OccidentalLifeTribute, _FexProducts.OccidentalLifeInnovativeSolutions, _FexProducts.OccidentalLifeSeniorChoice, _FexProducts.OccidentalLifeClearChoice, _FexProducts.OccidentalLifeGoldenSolution, _FexProducts.OccidentalLifePlatinumSolutionLegacyPlan,), states=()),
    'oxford-life': ProductCarrierMetadata(display_name='Oxford Life', products=(_FexProducts.OxfordLifeSimplifiedIssue,), states=()),
    'pekin': ProductCarrierMetadata(display_name='Pekin', products=(_FexProducts.PekinWholeLife,), states=()),
    'pioneer-american': ProductCarrierMetadata(display_name='Pioneer American', products=(_FexProducts.PioneerAmericanIndependentAmerican, _FexProducts.PioneerAmericanNorthstarLegacy,), states=()),
    'prosperity': ProductCarrierMetadata(display_name='Prosperity', products=(_TermProducts.ProsperityFamilyFreedomTerm,), states=()),
    'protective-life': ProductCarrierMetadata(display_name='Protective Life', products=(_TermProducts.ProtectiveLifeCustomChoiceTerm, _TermProducts.ProtectiveLifeClassicChoiceTerm,), states=()),
    'prudential': ProductCarrierMetadata(display_name='Prudential', products=(_TermProducts.PrudentialEssentialTermValue, _TermProducts.PrudentialEssentialTermPlus,), states=()),
    'royal-arcanum': ProductCarrierMetadata(display_name='Royal Arcanum', products=(_FexProducts.RoyalArcanumGraded, _FexProducts.RoyalArcanumSimplifiedIssue,), states=()),
    'royal-neighbors': ProductCarrierMetadata(display_name='Royal Neighbors', products=(_FexProducts.RoyalNeighborsEnsuredLegacy,), states=()),
    's-usa': ProductCarrierMetadata(display_name='S.USA', products=(_FexProducts.SUsaGoldenPromise,), states=()),
    'sagicor': ProductCarrierMetadata(display_name='Sagicor', products=(_TermProducts.SagicorSageTerm,), states=()),
    'sbli': ProductCarrierMetadata(display_name='SBLI', products=(_FexProducts.SbliLivingLegacy, _TermProducts.SbliTTerm,), states=()),
    'securico-life': ProductCarrierMetadata(display_name='Securico Life', products=(_FexProducts.SecuricoLifeFinalExpense,), states=()),
    'security-national': ProductCarrierMetadata(display_name='Security National', products=(_FexProducts.SecurityNationalSimpleSecurity,), states=()),
    'senior-life': ProductCarrierMetadata(display_name='Senior Life', products=(_TermProducts.SeniorLifeTermLife, _FexProducts.SeniorLifeWholeLife,), states=()),
    'sentinel-security': ProductCarrierMetadata(display_name='Sentinel Security', products=(_FexProducts.SentinelSecurityNewVantage,), states=()),
    'sons-of-norway': ProductCarrierMetadata(display_name='Sons of Norway', products=(_FexProducts.SonsOfNorwayLegacySure, _FexProducts.SonsOfNorwayWholeLife,), states=()),
    'transamerica': ProductCarrierMetadata(display_name='TransAmerica', products=(_FexProducts.TransamericaFeExpressSolution, _TermProducts.TransamericaTrendsetterSuper, _TermProducts.TransamericaTrendsetterLb, _FexProducts.TransamericaSolution,), states=()),
    'trinity': ProductCarrierMetadata(display_name='Trinity', products=(_FexProducts.TrinityGoldenEagle,), states=()),
    'united-farm-and-family': ProductCarrierMetadata(display_name='United Farm And Family', products=(_FexProducts.UnitedFarmAndFamilyWholeLife,), states=()),
    'united-home-life': ProductCarrierMetadata(display_name='United Home Life', products=(_FexProducts.UnitedHomeLifeWholeLife,), states=()),
    'william-penn': ProductCarrierMetadata(display_name='William Penn', products=(_TermProducts.WilliamPennOpterm,), states=()),
}

_ALL_CARRIERS: tuple[str, ...] = ('aetna', 'aetna-accendo', 'aflac', 'american-amicable', 'american-home-life', 'americo', 'ameritas', 'baltimore-life', 'banner', 'betterlife', 'centrian', 'cica-life', 'cigna', 'combined', 'corebridge', 'emc', 'everest-ia-american', 'family-benefit-life', 'fidelity-life', 'first-guaranty-insurance', 'foresters', 'gerber', 'global-atlantic', 'gpm', 'gpm-life', 'gtl', 'hero-life', 'illinois-mutual', 'john-hancock', 'kansas-city-life', 'kskj', 'liberty-bankers', 'life-shield', 'lincoln', 'manhattan-life', 'mutual-of-omaha', 'nationwide', 'newbridge', 'north-american', 'occidental-life', 'oxford-life', 'pekin', 'pioneer-american', 'prosperity', 'protective-life', 'prudential', 'royal-arcanum', 'royal-neighbors', 's-usa', 'sagicor', 'sbli', 'securico-life', 'security-national', 'senior-life', 'sentinel-security', 'sons-of-norway', 'transamerica', 'trinity', 'united-farm-and-family', 'united-home-life', 'william-penn')


class _ProductCarriersAPI:
    """Catalog API for carriers."""

    __slots__ = ()

    def values(self) -> tuple[str, ...]:
        return _ALL_CARRIERS

    def metadata(self, c: str) -> ProductCarrierMetadata:
        m = _CARRIERS.get(c)
        if m is None:
            raise KeyError(f"ProductCarriers.metadata: unknown carrier {c!r}")
        return m


ProductCarriers = _ProductCarriersAPI()
